picframe/viewer_display.py

```diff
@@ -358,7 +359,16 @@ def __make_text(self, pic, paused, side=0, pair=False):
                 fdt = time.strftime(self.__show_text_fm, time.localtime(pic.exif_datetime))
                 info_strings.append(fdt)
             if (self.__show_text & 16) == 16 and pic.location is not None: # location
-                info_strings.append(pic.location) #TODO need to sanitize and check longer than 0 for real
+                location = pic.location
+                # search for and remove substrings from the location text
+                if self.__geo_suppress_list is not None:
+                    for part in self.__geo_suppress_list:
+                        location = location.replace(part, "")
+                # remove any redundant concatination strings once the substrings have been removed
+                location = location.replace(" ,", "")
```

Comment:
Shouldn't this be part of the if branch?

Reply:
Having it outside of the if branch means that the tidy up is only performed once.

Thinking about it if the values in the list had been crafted as "somewhere, " for parts within a location then it's possible that the replacement won't find any redundant concatination symbols.

Reply:
I guess the code snippet was to small.

I talked about ` if self.__geo_suppress_list is not None:` 

````python
if (self.__show_text & 16) == 16 and pic.location is not None: # location
  location = pic.location
  # search for and remove substrings from the location text
  if self.__geo_suppress_list is not None:
      for part in self.__geo_suppress_list:
          location = location.replace(part, "")
  # remove any redundant concatination strings once the substrings have been removed
  location = location.replace(" ,", "")
  # remove any trailing commas or spaces from the location
  location = location.strip(", ")
  info_strings.append(location) #TODO need to sanitize and check longer than 0 for real
````
Tidy up if and only if there is something to suppress.

````python
if (self.__show_text & 16) == 16 and pic.location is not None: # location
  location = pic.location
  # search for and remove substrings from the location text
  if self.__geo_suppress_list is not None:
      for part in self.__geo_suppress_list:
          location = location.replace(part, "")
    # remove any redundant concatination strings once the substrings have been removed
    location = location.replace(" ,", "")
    # remove any trailing commas or spaces from the location
    location = location.strip(", ")
  info_strings.append(location) #TODO need to sanitize and check longer than 0 for real
````

Reply:
Yes that'd be better